src/OptimalTransport.jl

```diff
@@ -112,6 +116,109 @@ function emd2(μ, ν, C, optimizer)
     return dot(γ, C)
 end
 
+
+"""
+w2mvncost(μ::MvNormal, ν::MvNormal)
+
+Compute the 2-Wassertein distance between Multivariate Normal distributions.
+Note that for two Normal distributions
+``\\mu \\in N({m}_\\mu,{\\Sigma}_\\mu)`` and
+``\\nu \\in N({m}_\\nu,{\\Sigma}_\\nu)``
+the 2-Wassertein distance has a closed form given by:
+```math
+  W_2(\\mu,\\nu) = \\left[|m_\\mu - m_\\nu|^2 + tr\\left(
+  \\Sigma_\\mu +
+  \\Sigma_\\nu -
+  2 \\left(
+    \\Sigma_\\mu^{1/2} \\Sigma_\\nu
+    \\Sigma_\\mu^{1/2}
+    \\right)^{1/2}
+  \\right) \\right]^{1/2}
+```
+where,
+```math
+  A =
+  \\Sigma_\\mu^{-1/2}\\left(
+  \\Sigma_\\mu^{1/2} \\Sigma_\\nu
+  \\Sigma_\\mu^{1/2}
+  \\right)^{1/2}
+  \\Sigma_\\mu^{-1/2}.
+```
+"""
+function w2mvncost(X::MvNormal, Y::MvNormal)
+    XΣsqrt = X.Σ^(1/2)
+    w2     = sqrt(norm(X.μ - Y.μ)^2 + tr(X.Σ + Y.Σ - 2*(XΣsqrt*Y.Σ*XΣsqrt)^(1/2)))
+    return w2
```

Comment:
I think it is better to use dedicated functions for the Bures metric that allow optimizations based on the type of the covariance matrix. For many cases the computations here will be quite inefficient.

You can have a look at how I implemented it in https://github.com/devmotion/CalibrationErrorsDistributions.jl/blob/85d8ca7a71b319b31d64b520aff344fd6c6758e6/src/distances/wasserstein.jl#L28 and https://github.com/devmotion/CalibrationErrorsDistributions.jl/blob/main/src/distances/bures.jl.

Reply:
Great! I was wondering if there was a more efficient way of doing this, but I did not find this implementation on POT.py.
I'll take a look at your code and modify.

Reply:
I see that use used `PDMats.jl` in the calculations for the Bures metric. So you suggests we also add that as a dependency?

Reply:
Distributions already depends on it :shrug:

Reply:
Ok then. I saw that you've implemented various cases already, so I'll just reuse your code on for the Bures calculation.